src/RoslynPad.Common.UI/ViewModels/MainViewModel.cs

```diff
@@ -432,6 +443,46 @@ public async Task OpenFile()
 
         OpenDocument(document);
     }
+    public async Task OpenFolder()
+    {
+        if (!IsInitialized) return;
+
+        var dialog = _serviceProvider.GetRequiredService<IOpenFolderDialog>();
+        //dialog.InitialDirectory = "C:\\Users\\luxus\\source\\repos\\roslynpad\\roslynpad\\src\\";
+        var folderPath = await dialog.ShowAsync().ConfigureAwait(true);
+
+        if (folderPath == null)
+        {
+            return;
+        }
+
+        if (!DocumentRoot.Path.Equals(folderPath, StringComparison.OrdinalIgnoreCase))
+        {
+            Settings.DocumentPath = folderPath;
+
+            DocumentRoot = SetDocumentRoot();
```

Comment:
I'm not sure this is the behavior the issue author intended - it's basically replacing the document panel with a different directory. Perhaps it would be better to add a "virtual" node to the existing tree with the imported documents.

Reply:
Also maybe we don't need to open all the documents, let the user decide which ones they want from the tree.